keyboards/neson_design/nico/nico.c

```diff
@@ -0,0 +1,86 @@
+/**
+ * @file nico.c
+ *
+    Copyright 2023 astro
+
+    This program is free software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "quantum.h"
+#ifdef RGBLIGHT_ENABLE
+#include "rgblight.h"
+
+static bool alert = false;
+static bool backup = false;
+static LED_TYPE caps_led;
+static uint16_t last_ticks = 0;
+
+#define ALERT_INTERVAL      500
+#define ALERM_LED_R     0xFF
+#define ALERM_LED_G     0xA5
+#define ALERM_LED_B     0x00
+//golden 0xFF, 0xD9, 0x00
+
+void housekeeping_task_kb(void)
+{
+    if (host_keyboard_led_state().caps_lock) {
+        if (!backup) {
+            caps_led.r = led[4].r;
+            caps_led.g = led[4].g;
+            caps_led.b = led[4].b;
+            backup = true;
+        }
+        if(alert) {
+            led[4].r = ALERM_LED_G;
+            led[4].g = ALERM_LED_R;
+            led[4].b = ALERM_LED_B;
+        } else {
+            led[4].r = 0;
+            led[4].g = 0;
+            led[4].b = 0;
+        }
+        if (timer_elapsed(last_ticks) > ALERT_INTERVAL) {
+            alert = !alert;
+            last_ticks = timer_read();
+        }
+        ws2812_setleds(led, RGBLED_NUM);
+    } else {
+        if (backup) {
+            led[4].r = caps_led.r;
+            led[4].g = caps_led.g;
+            led[4].b = caps_led.b;
+            backup = false;
+            ws2812_setleds(led, RGBLED_NUM);
+        }
+    }
+    housekeeping_task_user();
+}
+
+void rgblight_call_driver(LED_TYPE *start_led, uint8_t num_leds)
```

Comment:
What is the purpose of this?

Reply:
there are two led bars on left and right, sync them with same led colors.